### ISSUE TITLE
 screenshots has gone haywire notification on startup

### ISSUE THREAD

**Issue by wresuolc** [mozilla-services/screenshots#2902]
Filed from https://bugzilla.mozilla.org/show_bug.cgi?id=1368401

Multiple reports of "Screenshots has gone haywire" when the browser is just starting up (no screenshot being taken).

/cc @SoftVision-PaulOiegas @SoftVision-CosminMuntean

**Comment by autonome** [comment]
I'm on Mac. I see this at every browser start. I've never touched any pref related to the feature, and when I search about:config for the pref you mentioned, there's nothing matching it there.

**Comment by mikeconley** [comment]
Ditto as @autonome. I think I can reproduce this pretty easily, if there's some kind of logging you want me to gather or a breakpoint you want me to set.

**Comment by mikeconley** [comment]
<img width="371" alt="screen shot 2017-05-29 at 10 27 20 am" src="https://cloud.githubusercontent.com/assets/195042/26554530/93aecb0c-445e-11e7-8fa8-4687b84cbcf4.png">

To be clear, this is what I've been seeing.

**Comment by SoftVision-CosminMuntean** [comment]
I haven't managed to reproduce this issue on Mac OS 10.12 and also on Windows 7 x64.

@autonome can you please give us more information?
Did you had Screenshots enabled before it came enabled by default? Have you managed to save or share shots? Or you have not used it at all?
Also, which Mac OS version are you using?

**Comment by philipp-sumo** [comment]
i think i'm seeing something similar on each nightly startup (win 10 here).
at the bottom of the notification it's saying "TypeError: registrationInfo is undefined"
![2017-05-29 17_46_10-](https://cloud.githubusercontent.com/assets/1524147/26556004/449d4928-4498-11e7-830b-33db1c7128e5.png)
![2017-05-29 17_54_57-browser-konsole](https://cloud.githubusercontent.com/assets/1524147/26556005/44c0c98e-4498-11e7-9495-ef9c3b7f3927.png)

**Comment by SoftVision-CosminMuntean** [comment]
@autonome @philipp-sumo can you guys please retest this issue using a new clean profile and see if is still reproducible?
Also, can you attach the list with all your add-ons installed?

**Comment by philipp-sumo** [comment]
i think the issue is in a profile where i had installed the screenshot testpilot addon in the past.

i can still reproduce the issue in a new profile where https://testpilot.firefox.com/experiments/page-shot is installed next to the system addon.

**Comment by past** [comment]
That is my case as well. I had the test pilot version installed and then nightly updated to a version that flips the pref and now I see this error even after uninstalling the test pilot version.

**Comment by autonome** [comment]
Yep I have​ Test Pilot installed as well. Either way, new profile isn't the
right test-case as nearly 100% of users executing these code paths will be
pre-existing users on a Firefox upgrade path. Mostly not with TP of course.

**Comment by SoftVision-CosminMuntean** [comment]
@philipp-sumo thanks!  You're right.
I can reproduce this issue on all Firefox profiles where I had Page Shot installed from Test Pilot.

This is also reproducible if you enable Page Shot from Test Pilot and you currently have Screenshots enabled.
[Steps to reproduce]:
1. Open latest Nightly (55.0a1, Build ID: 20170529030204) build.
2. Navigate to "https://testpilot.firefox.com/" page.
3. Click the "Page Shot" experiment card and click the "Install Test Pilot & Enable Page Shot" add-on.
4. Restart the browser.

[Expected result]:
- No errors are displayed.

[Actual results]:
- A "Whoa! Firefox Screenshots went haywire" error is displayed.

[Notes]:
- Here is a screen recording with the issue: https://goo.gl/P2Q1GQ
- Here is a screenshot with browser console errors:
![error](https://cloud.githubusercontent.com/assets/19226534/26556812/7c10076c-44a5-11e7-91d7-0bc4a5b58c07.png)

**Comment by rnewman** [comment]
Same as #2927.

**Comment by johngruen** [comment]
Report from a Mac user on nightly with the following errors in browser console:
```
Unhandled error: <unavailable> undefined  catcher.js:15
Unhandled error: Object { fromMakeError: true, name: “ReferenceError”, message: “ReferenceError: ui is not defined”, stack: “mousedown@moz-extension://4fa867f3-…” } undefined  catcher.js:15:5
Unhandled error: <unavailable> undefined catcher.js:15
Unhandled error: Object { fromMakeError: true, name: “ReferenceError”, message: “ReferenceError: ui is not defined”, stack: “mousedown@moz-extension://4fa867f3-…” } undefined  catcher.js:15:5
Unhandled error: Object { fromMakeError: true, name: “ReferenceError”, message: “ReferenceError: ui is not defined”, stack: “mousedown@moz-extension://4fa867f3-…” } undefined  catcher.js:15:5
Unhandled error: Object { fromMakeError: true, name: “ReferenceError”, message: “ReferenceError: ui is not defined”, stack: “mousedown@moz-extension://4fa867f3-…” } undefined  catcher.js:15:5
ReferenceError: ui is not defined
```
User says the error is confined to "a bad tab" possble the webExtension has crashed?

**Comment by ianb** [comment]
That last one might be https://github.com/mozilla-services/screenshots/issues/2838

**Comment by johngruen** [comment]
@ianb yeah this was called out on Slack by JP on the PI team. He says he had previously used the tab to take a screenshot

**Comment by mozillamarcia** [comment]
I see this frequently on Windows Surface Pro, even after the browser has been running. Usually I am trying to do something on a news site such as cnn.com or nypost.com, and it happens the first time I try to take a screenshot.

**Comment by SoftVision-CosminMuntean** [comment]
@mystor can you please navigate to "about:support" page and see what Firefox Screenshots version do you have?
The Screenshots version is displayed in the "Nightly Features" section.

**Comment by rnewman** [comment]
Same here. 8.1.0.

**Comment by SoftVision-CosminMuntean** [comment]
Strange, I have the latest Nightly (2017-06-07, Build ID: 20170607030206) build and the Screenshots version is 9.0.0.  I haven't managed to reproduce this issue on this version.

**Comment by mystor** [comment]
> @mystor can you please navigate to "about:support" page and see what Firefox Screenshots version do you have?
> The Screenshots version is displayed in the "Nightly Features" section. 

Firefox Screenshots 8.1.0 - I just updated my nightly version right before made that comment.
Firefox build ID 20170606100219, so it's one version out of date (I suppose it downloaded the new version yesterday and just updated to that).

**Comment by mozillamarcia** [comment]
On my Windows Surface Pro, using the latest nightly and Version 9.0.0 of screenshots, I still seem to see it the first time I try to use the feature.

**Comment by ianb** [comment]
We fixed some of these issues, but we aren't sure where the remaining issues are coming from.

@mozillamacia: if you see this, can you look in your Browser Console to see what the exception is? Instructions on how to open it [can be found here](https://developer.mozilla.org/en-US/docs/Tools/Browser_Console#Opening_the_Browser_Console) – we're interested in the pink errors, like what was in comment https://github.com/mozilla-services/screenshots/issues/2902#issuecomment-304697806

**Comment by mozillamarcia** [comment]
![img_20170608_142506](https://user-images.githubusercontent.com/2171777/26944606-2edceaa0-4c57-11e7-8130-8dc90f9db680.jpg)

Hopefully you can see this - I took it with my camera - several attempts to capture it with Windows tools failed.

**Comment by ianb** [comment]
Thanks, that's super helpful!

**Comment by 6a68** [comment]
PR #3011 adds a 5-second startup delay before showing any error notifications. The errors will still be sent to Sentry, and we can whittle away initialization edge case bugs behind the scenes - without annoying users.

**Comment by wresuolc** [comment]
This should be fixed in 10.2.0.  That's in line to land in bugzilla.mozilla.org/show_bug.cgi?id=1372310

Thanks!
